wgengine/userspace_test.go

```diff
@@ -181,21 +183,24 @@ func TestUserspaceEnginePortReconfig(t *testing.T) {
 		},
 	}
 	routerCfg := &router.Config{}
-	if err := ue.Reconfig(cfg, routerCfg, &dns.Config{}, nil); err != nil {
+	if err := ue.Reconfig(cfg, routerCfg, &dns.Config{}); err != nil {
 		t.Fatal(err)
 	}
 	if got := ue.magicConn.LocalPort(); got != startingPort {
 		t.Errorf("no debug setting changed local port to %d from %d", got, startingPort)
 	}
-	if err := ue.Reconfig(cfg, routerCfg, &dns.Config{}, &tailcfg.Debug{RandomizeClientPort: true}); err != nil {
+
+	randomizeClientPort = func() bool { return true }
+	if err := ue.Reconfig(cfg, routerCfg, &dns.Config{}); err != nil {
 		t.Fatal(err)
 	}
 	if got := ue.magicConn.LocalPort(); got == startingPort {
 		t.Errorf("debug setting did not change local port from %d", startingPort)
 	}
 
 	lastPort := ue.magicConn.LocalPort()
-	if err := ue.Reconfig(cfg, routerCfg, &dns.Config{}, nil); err != nil {
+	randomizeClientPort = func() bool { return false }
```

Comment:
use tstest.Replace so that it gets unset after the test is finished. Otherwise we will spend hours debugging a test failure only to discover this :P

Reply:
I did at the top already.

Reply:
oh i see, sure